src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -15,18 +16,25 @@
 
 (trapperkeeper/defservice jruby-puppet-pooled-service
                           jruby/JRubyPuppetService
-                          [[:ConfigService get-config]
-                           [:ShutdownService shutdown-on-error]
-                           [:PuppetProfilerService get-profiler]
-                           [:PoolManagerService create-pool]]
+                          {:required [[:ConfigService get-config]
+                                      [:ShutdownService shutdown-on-error]
+                                      [:PuppetProfilerService get-profiler]
+                                      [:PoolManagerService create-pool]]
+                           :optional [MetricsService]}
```

Comment:
@jpinsonault You can toggle whether to report metrics to jmx or not, as well as whether to report metrics to graphite or not.. You can also enable/disable the profiler. I was thinking that this was maybe kind of akin to the profiler - a couple folks I've talked to have reported that they were maybe seeing some performance issues with that, so they turned it off. I would be concerned that if we might run into some performance issues with these metrics and so we would want a way to turn them off in case someone upgraded and ran into performance issues.